### PR TITLE
catalog of my things

### PR DESCRIPTION
**Pull request by Naledi-Dikgale**
### In this Pull Request we did the following

- [ ] Create a `main.rb` file that will serve as our console app entry-point and at the start it:
- Present the user with a list of options to perform.
- Let users choose an option.
- If needed, ask for parameters for the option.
- Have a way to quit the app.

- [ ] Implemented all methods visible in the diagram.
- [ ] All data should be preserved by saving collections in `.json` files.
- [ ] Wrote unit tests for all methods in all classes.
- [ ] Recorded a project walkthrough video.

### CODE REVIEW

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @Naledi-Dikgale , @Patfarmurs , and @citec-47         👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️  
- Nice work with the setup✔️
- Nice test coverage ✔️
- Good work with SQL ✔️ 
- Nice work with modular codebase       
  
# Required Changes  🌵  

- [ ] I've made some highlights on the chnages required in the comment section. Kindly review and make the appropriate chnages 👍 
- [ ] As a major note, please make sure you stick to the database values as much as possible 👍 

_Check the comments for a detailed review 

# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: schema/schema.sql [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  FOREIGN KEY (label_id) REFERENCES label(id)
+);
+
+-- Create genres table
+CREATE TABLE Genres (
+  id INT Generated ALWAYS AS IDENTITY PRIMARY KEY,
+  name VARCHAR(100),
+  foreign key (item_id) references item(id)
+);
+
```
Comment (danmainah) [inline schema/schema.sql:22]: - [ ] @Naledi-Dikgale , @Patfarmurs, and @citec-47 Great work implementing the changes requested, however, you forgot to add an items array in your Genres table to match the database UML gave as previously requested, this also applies to the Author and Label tables.Kindly consider implementing theses changes.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Naledi-Dikgale , @Patfarmurs and @citec-47 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Good design- implemented all the required menu items and ensured they are all working.
✔️  Professional written README file with all sections required.


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @Naledi-Dikgale , @Patfarmurs and @citec-47 ,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️ Implemented changes requested to ensure the schema matches the database UML.


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**